### PR TITLE
[Story] CRUD invitation templates

### PR DESCRIPTION
**Pull request by neilboyd**
Implement functionality to create/update/delete invitation templates

### CODE REVIEW

File: Tests/Services/NfieldSurveyInvitationTemplatesServiceTests.cs [hunk 1]
```diff
@@ -71,18 +74,202 @@ public void TestGetAsync_SurveyHasInvitationTemplates_ReturnsInvitationTemplates
...
+                UnwrapAggregateException(target.AddAsync(null, null)));
+        }
+
+        [Fact]
+        public void TestAddAsync_SurveyIdIsEmpty_Throws()
+        {
+            var target = new NfieldSurveyInvitationTemplatesService();
+
+            Assert.Throws<ArgumentException>(() =>
+                UnwrapAggregateException(target.AddAsync("  ", null)));
```
Comment (pilarodriguez) [inline Tests/Services/NfieldSurveyInvitationTemplatesServiceTests.cs]: It is testing a SurveyId with whitespaces, not empty

File: Tests/Services/NfieldSurveyInvitationTemplatesServiceTests.cs [hunk 2]
```diff
@@ -71,18 +74,202 @@ public void TestGetAsync_SurveyHasInvitationTemplates_ReturnsInvitationTemplates
...
+                UnwrapAggregateException(target.UpdateAsync(null, null)));
+        }
+
+        [Fact]
+        public void TestUpdateAsync_SurveyIdIsEmpty_Throws()
+        {
+            var target = new NfieldSurveyInvitationTemplatesService();
+
+            Assert.Throws<ArgumentException>(() =>
+                UnwrapAggregateException(target.UpdateAsync("  ", null)));
```
Comment (pilarodriguez) [inline Tests/Services/NfieldSurveyInvitationTemplatesServiceTests.cs]: Same here

File: Tests/Services/NfieldSurveyInvitationTemplatesServiceTests.cs (hunk 1 above)
Comment (maartje) [inline Tests/Services/NfieldSurveyInvitationTemplatesServiceTests.cs]: fixed by changing the name of the test

File: Tests/Services/NfieldSurveyInvitationTemplatesServiceTests.cs (hunk 2 above)
Comment (maartje) [inline Tests/Services/NfieldSurveyInvitationTemplatesServiceTests.cs]: fixed by changing the name of the test

**Pull request merged**